lib/ssh/setup-ssh-agent

```diff
@@ -0,0 +1,9 @@
+sshdir=$(getent passwd $USER | cut -d: -f6)/.ssh
+mkdir $sshdir
+chmod 700 $sshdir
+cat > $sshdir/key
+chmod 600 $sshdir/key
+eval "$(ssh-agent -s)"
+gha-set-env SSH_AUTH_SOCK <<< $SSH_AUTH_SOCK
+gha-set-env SSH_AGENT_PID <<< $SSH_AGENT_PID
+ssh-add $sshdir/key
```

Comment:
The `gha` commands in this script means its scope is smaller than ssh. I think it belongs best under github-actions?

All together:

`lib/github-actions/ssh-agent`
```suggestion
#!/bin/bash
set -euxo pipefail
eval "$(ssh-agent)"
ssh-add -  # private key on stdin
gha-set-env SSH_AUTH_SOCK <<< "$SSH_AUTH_SOCK"
gha-set-env SSH_AGENT_PID <<< "$SSH_AGENT_PID"
```

`bin/gha-ssh-agent -> ../lib/github-actions/ssh-agent`

Reply:
adopted this organization.